recipe/meta.yaml

```diff
@@ -54,23 +55,29 @@ outputs:
         - perl  # [osx]
         - autoconf  # [osx]
         - automake  # [osx]
-        - libtool   # [unix]
+        - libtool  # [unix]
         - make  # [unix]
       host:
         #- openpmix
         #- prrte
-        - libhwloc
         - libevent
+        - libhwloc
         - libnl  # [linux]
+        - libnuma  # [linux and not ppc64le]
+        - libudev1  # [linux and not ppc64le]
+        - numactl  # [linux and not ppc64le]
+        - rdma-core  # [linux and not ppc64le]
```

Comment:
Q: are `libnuma` and `rdma-core` really needed at build time? I see no linkage reported 
```
...
2024-09-24T12:11:46.3930863Z WARNING (openmpi): dso library package conda-forge/linux-64::libnuma==2.0.18=h4ab18f5_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
...
2024-09-24T12:11:46.3933755Z WARNING (openmpi): run-exports library package conda-forge/linux-64::rdma-core==53.0=he02047a_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
...
```

cc @dalcinl @pentschev @marcinz in case you can shed some light here

Reply:
I wonder if those are requirements that are only needed by UCX, I don't know anything about them in OpenMPI though. I know [libnuma's dependency was removed from UCX quite some time ago](https://github.com/openucx/ucx/pull/8698), and rdma-core is used to provide InfiniBand support, which I don't know if OpenMPI has native support for (i.e., without depending on UCX).

Reply:
These dependencies may not be mandatory at build time for our core functionality.
They support specific advanced features such as NUMA awareness and RDMA capabilities. 
I guess that users interested in these optimizations can manually load the respective libraries if required.

Reply:
@j34ni would you mind updating the PR title/description to reflect the above discussion and the latest changes?

Reply:
Is that better?

Reply:
Yeah thanks!